addons/mail/models/res_partner.py

```diff
@@ -82,7 +87,16 @@ def _get_needaction_count(self):
     # ------------------------------------------------------------
 
     def _mail_get_partners(self, introspect_fields=False):
-        return dict((partner.id, partner) for partner in self)
+        return {partner.id: partner for partner in self}
+
+    @api.depends('_starred_message_ids')
+    def _compute_starred_message_ids(self):
+        for record in self:
+            record.starred_message_ids = record.sudo()._starred_message_ids.sudo(False)._filtered_access('read')
```

Comment:
I guess technically it should reset to the `su` state to the one of `record.env` rather than always `False` in case we want/need to sudo-read the compute.

Reply:
I thought about `with_env(self.env)`.If you prefer, I can change it.

Now, I am wondering if it should be something like `records.sudo()._starred_message_ids.with_user(record.user_ids[:1] or self.env.uid)._filtered_access('read')` to retrieve only the messages that the partner can see.

Reply:
I don't think we ever need to read the value for another user than the current user in practice, but yes good catch, it should be more like what you suggest.

And if we need to get to that, then it should probably be a field on user model rather than on partner.

Honestly this field is rather useless, it's used so little that if we need such a hack and a duplicate field, we might as well make a simple method for returning the values. But even that we don't do?

The field is only used for writing on it?

Reply:
There: https://github.com/odoo/odoo/pull/219282
field gone = problem solved